### ISSUE TITLE
DllNotFoundException: Unable to load DLL 'sni.dll': Access is denied.

### ISSUE THREAD

**Issue by steveoh** [dotnet/cli#5262]
## Steps to reproduce
`dotnet public -c Production`
copy files to a windows 2012 r2 iis8.5 machine with the hosting bundle installed.

```
λ dotnet --version

Microsoft .NET Core Shared Framework Host

  Version  : 1.1.0
  Build    : 928f77c4bc3f49d892459992fb6e1d5542cb5e86
```

## Expected  behavior

I expect the application to work when published to IIS when using a sql client like it does when run on OSX.

## Actual behavior

```
An unhandled exception occurred while processing the request.

DllNotFoundException: Unable to load DLL 'sni.dll': Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
System.Data.SqlClient.SNINativeMethodWrapper.SNIInitialize(IntPtr pmo)

TypeInitializationException: The type initializer for 'System.Data.SqlClient.SNILoadHandle' threw an exception.
System.Data.SqlClient.TdsParser..cctor()

TypeInitializationException: The type initializer for 'System.Data.SqlClient.TdsParser' threw an exception.
System.Data.SqlClient.TdsParser..ctor(bool MARS, bool fAsynchronous)
```

## Environment data
`dotnet --info` output:

```
$ dotnet --info
.NET Command Line Tools (1.0.0-preview2-1-003177)

Product Information:
 Version:            1.0.0-preview2-1-003177
 Commit SHA-1 hash:  a2df9c2576

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
```

**Comment by rjperes** [comment]
I'm having the same problem on Windows 7. Never happened before.

**Comment by schellap** [comment]
It looks like a pre-req issue. Can you confirm if you have the pre-reqs installed on Windows as described here? https://docs.microsoft.com/en-us/dotnet/articles/core/windows-prerequisites#net-core-dependencies

Esp., the ucrt update.

**Comment by schellap** [comment]
And this KB: For Windows 7 and Windows Server 2008 machines only:
Make sure that your Windows installation is up-to-date and includes hotfix KB2533623 installed through Windows Update.

**Comment by steveoh** [comment]
I am on windows server 2012 r2 so I will skip the windows 7 and 2008 instructions

**Comment by steveoh** [comment]
I have these vc++ items

![image](https://cloud.githubusercontent.com/assets/325813/21878091/fe33c92e-d84c-11e6-867b-40c8f21c9712.png)

[KB2533623](https://www.microsoft.com/en-us/download/details.aspx?id=26767) looks like another windows 7 update that was published in 2011. 

My update history

![image](https://cloud.githubusercontent.com/assets/325813/21878134/55fc2ce6-d84d-11e6-84b4-49a91f6ccbe9.png)

**Comment by steveoh** [comment]
What is the ucrt update @schellap?

**Comment by schellap** [comment]
@steveoh I think you have the necessary pre-reqs. I got misled by the win7 comment above.

Can you guide me through how to publish your app from your repo -- what steps are you using?

I did the following:
```
cd api
dotnet restore
dotnet publish -c Production
```

Is the above your flow?

**Comment by steveoh** [comment]
Yes basically. [this issues](https://github.com/aspnet/IISIntegration/issues/315) explains it a bit more when I thought it was an IIS integration issue.

**Comment by piotrpMSFT** [comment]
Any updates here?

**Comment by steveoh** [comment]
None from me

**Comment by azisoft** [comment]
Had the same issue. Fixed it by assigning full control to %USERPROFILE%\.nuget. You can also try to change  the app pool identity in IIS to LocalSystem.

**Comment by keycad** [comment]
Thanks @azisoft , I've just solved the issue following your second advice: change the app pool identity to LocalSystem

**Comment by MongkonEiadon** [comment]
![image](https://user-images.githubusercontent.com/3174809/66902865-9ffe1680-f02b-11e9-9c4e-8b27395beba9.png)

It's work for me, thanks for saving my time

**Comment by RomBrz** [comment]
This happen with me.
Giving more permissions to the Application Pool user for me is the worst option, for God's sake! For who don't know, the Application Pool Identity was created to give more security to the entire server!!

https://docs.microsoft.com/en-us/iis/manage/configuring-security/application-pool-identities

All my sites run with Application Pool Identity, with "Read" only permission to IISAppPool\APPLICATIONPOOLNAME to the site's folder.

With .NET Core i'm having this problem with some sites, that could be solved by giving ONLY the folder's DLL the "Read and Execute" permission solve the problem.
(In that case of "sni.dll", giving "Read and Execute" to [...]\runtimes\win-arm64\native should work). If you don't want to debug to the correct way, this is, in my opinion, by far, best than giving "System" permission to the App Pool)

But it's not how should be, in my opinion, as "Execute" permission give a security flaw as someone could inject a code in that folder and execute remotely.

Something is making the AppPoolIdentity need to execute the DLLs needed to access SQL Server.
